keps/sig-network/1797-configure-fqdn-as-hostname-for-pods/README.md

```diff
@@ -0,0 +1,762 @@
+-   Create a pod with fqdInHostname: true and a subdomain in its spec, and check the value of hostname field in the kernel
+
+
+### Graduation Criteria
+-   Compatible with major systems (e.g. linux, windows)
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real world usage
+- N installs
+- More rigorous forms of testing e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least 2 releases between beta and
+GA/stable, since there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a deprecated flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+-->
+
```

Comment:
Maybe just mention that we'll gate it off for one release (I think you're shooting for 19) and then enable it as GA in the next.  Or do you think Beta is needed?

We can only GA it when we know the Windows story.

Reply:
Agree, probably no need for Beta.

+1 on knowing expected behavior for Windows

Reply:
The feature gate is mainly for cluster upgrade/downgrade scenarios to enforce API consistency correct?
I thought it was recommended to always follow a 3 release cycle for feature gates (API fields dropped by kubelet by default, API fields honored by kubelet by default, API fields set always honor API fields).

From a Windows stand-point I'm ok with skipping Beta - I just want to make sure we follow best practices.
@liggitt can you comment on this?

Reply:
It looks like it should be OK to skip beta per https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-field-in-existing-api-version

Reply:
from a data preservation perspective, only two releases are necessary (roll out the type change in one release, allow data into the new field in create and adding data in update requests in the next release)